.github/workflows/visualize.yml

```diff
@@ -0,0 +1,177 @@
+#        if: steps.cache-must.outputs.cache-hit != 'true'
+        run: |
+              bash ./scripts/MUST/install_tool.sh
+# setup env or running
+      - run: mkdir experiment
+      - name: Run CorrBench
+        run: |
+           export PATH=$PATH:${GITHUB_WORKSPACE}/MUST_install/bin && \
+           export LIBRARY_PATH=$LIBRARY_PATH:${GITHUB_WORKSPACE}/MUST_install/lib && \
+           export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${GITHUB_WORKSPACE}/MUST_install/lib && \
+           export CPATH=$CPATH:${GITHUB_WORKSPACE}/MUST_install/include && \
+           bash ${GITHUB_WORKSPACE}/scripts/tool_runner_CI.sh MUST
+      - uses: actions/upload-artifact@v2
+        with:
+          name: results_MUST
+          path: ${{ github.workspace }}/experiment/MUST/results_MUST.json
+
+
+
+
+  run-parcoach:
+    runs-on: ubuntu-latest
+    env:
+      MPI_CORRECTNESS_BM_DIR: ${{ github.workspace }}
+      MPI_CORRECTNESS_BM_EXPERIMENT_DIR: ${{ github.workspace }}/experiment
+    steps:
+      - uses: actions/checkout@v2
+# prerequisites
+      - run : sudo apt-get update && sudo apt-get install -y cmake git libopenmpi-dev clang-9 llvm-9 llvm-9-dev
+# install PARCOACH
+# building parcoach is quite fast anyway
+#      - name: Cache PARCOACH
+#        id: cache-parcoach
+#        uses: actions/cache@v2
+#        with:
+#          path: parcoach
+#          key: ${{ runner.os }}-cache-parcoach-master
+      #TODO BUILD
+      - name: Build PARCOACH
+#        if: steps.cache-parcoach.outputs.cache-hit != 'true'
+        run: |
+           bash ./scripts/PARCOACH/install_tool.sh
+#TODO provide a container with the installation of MUST
+# setup env or running
+      - run: mkdir experiment
+      - name: Run CorrBench
+        run: |
+           export PARCOACH_ROOT=${GITHUB_WORKSPACE}/parcoach
+           bash ${GITHUB_WORKSPACE}/scripts/tool_runner_CI.sh PARCOACH
+      - uses: actions/upload-artifact@v2
+        with:
+          name: results_PARCOACH
+          path: ${{ github.workspace }}/experiment/PARCOACH/results_PARCOACH.json
+
+
+  run-mpi-checker:
+    runs-on: ubuntu-latest
+    env:
+      MPI_CORRECTNESS_BM_DIR: ${{ github.workspace }}
+      MPI_CORRECTNESS_BM_EXPERIMENT_DIR: ${{ github.workspace }}/experiment
+    steps:
+      - uses: actions/checkout@v2
+# install clang-tidy
+      - run : sudo apt-get update && sudo apt-get install -y clang-tidy libopenmpi-dev
+# setup env or running
+      - run: mkdir experiment
+      - name: Run CorrBench
+        run: |
+           bash ${GITHUB_WORKSPACE}/scripts/tool_runner_CI.sh MPI-Checker
+      - uses: actions/upload-artifact@v2
+        with:
+          name: results_MPI-Checker
+          path: ${{ github.workspace }}/experiment/MPI-Checker/results_MPI-Checker.json
+
+
+  generate-visualization:
+    needs: [run-mpi-checker, run-parcoach, run-must]
+    runs-on: ubuntu-latest
+    env:
+      MPI_CORRECTNESS_BM_DIR: ${{ github.workspace }}
+#TODO automatically get these version numbers
```

Comment:
This is the third, or so, TODO w.r.t. the CI configuration and setup.
Should we have one CI-related issue that lists the current TODOs?

Reply:
yes this should be listed in the issues section